torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -0,0 +1,151 @@
+#include <torch/csrc/distributed/rpc/request_callback_impl.h>
+#include <c10/util/C++17.h>
+#include <torch/csrc/distributed/autograd/context/dist_autograd_container.h>
+#include <torch/csrc/distributed/autograd/context/dist_autograd_context.h>
+#include <torch/csrc/distributed/autograd/utils.h>
+#include <torch/csrc/distributed/rpc/future_message.h>
+#include <torch/csrc/distributed/rpc/python_rpc_handler.h>
+#include <torch/csrc/distributed/rpc/python_udf_call.h>
+#include <torch/csrc/distributed/rpc/python_udf_resp.h>
+#include <torch/csrc/distributed/rpc/rpc_with_autograd.h>
+#include <torch/csrc/distributed/rpc/rref.h>
+#include <torch/csrc/distributed/rpc/rref_context.h>
+#include <torch/csrc/distributed/rpc/script_call.h>
+#include <torch/csrc/distributed/rpc/script_remote_call.h>
+#include <torch/csrc/distributed/rpc/script_resp.h>
+#include <torch/csrc/distributed/rpc/script_rref_proto.h>
+#include <torch/csrc/distributed/rpc/utils.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+using namespace torch::distributed::autograd;
+
+std::unique_ptr<RpcCommandBase> RequestCallbackImpl::processRpc(
+    RpcCommandBase& rpc,
+    MessageType messageType) {
+  // TODO: RpcCommandBase should have an abstract execute() method that we can
+  // call here instead of having another switch statement here. Even better we
+  // could have abstract classes RpcRequest and RpcResp which inherit from
+  // RpBase and RpcRequest declares the abstract method execute() that we can
+  // call here. RpcResponse could have an abstract method to convert it to a
+  // python object.
+  switch (messageType) {
+    case MessageType::SCRIPT_CALL: {
```

Comment:
Suggestion: since we now have a class for each RpcCommand, we could expose a function

```std::unique_ptr<RpcCommandBase> RpcCommand::process()``` that would be called instead of this function processRpc -- eliminating the need for the switch, the static cast and having to pass MessageType after the Message object after the message got converted to command.

Reply:
Yes, this is exactly what I had in mind. I have a // TODO at the beginning of `processRpc` for exactly this. Didn't want to add that to this PR though.